Sources/TurboNavigationHierarchyController.swift

```diff
@@ -0,0 +1,209 @@
+import SafariServices
+import Turbo
+import UIKit
+import WebKit
+
+/// Handles navigation to new URLs using the following rules:
+/// https://github.com/joemasilotti/TurboNavigator#handled-flows
+class TurboNavigationHierarchyController {
+    let navigationController: UINavigationController
+    let modalNavigationController: UINavigationController
+
+    var rootViewController: UIViewController { navigationController }
+    var activeNavigationController: UINavigationController {
+        navigationController.presentedViewController != nil ? modalNavigationController : navigationController
+    }
+
+    var animationsEnabled: Bool = true
```

Comment:
Why does this exist? I don't see a way to set it publicly as a consumer of the library. If anything, I'd prefer to see this live in the path configuration somehow so individual links can be customized instead of a global option.

Reply:
Ha. That was a hack. There's no way to pass along a type of `UINavigationController` into this class anymore, so testing was broken. What I did was create this variable and keep it internal so consumers cannot use it but unit tests can reach it.

Reply:
Ah, nice catch! What about exposing the navigation controllers in the initializer but defaulted like we had before?

```swift
init(delegate: TurboNavigationHierarchyControllerDelegate, navigationController: UINavigationController = UINavigationController(), modalNavigationController: UINavigationController = UINavigationController()) {
    self.delegate = delegate
    self.navigationController = navigationController
    self.modalNavigationController = modalNavigationController
}
```

Reply:
Yeah, I think that's the way forward. I originally didn't go with this because I didn't want to expose them, but since this class is now internal, I don't see any issues at all.